packages/nimble/R/nimbleFunction_core.R

```diff
@@ -117,6 +117,30 @@ nimbleFunction <- function(setup         = NULL,
     return(generatorFunction)
 }
 
```

Comment:
This looks tricky! Could you describe in a comment how it's used? My guess is
```
## To substitute a user-provided .cpp file in place of a nimble generated .cpp file, first define
## a nimble function via nf <- nimbleFunction(...), and then set nf.specialHandling <- 'my_substitute.cpp'.
```
Maybe a simpler approach would be to just set a global `nimbleOptions(fakeCppFile = 'my_substitute.cpp')` and use this as the file in the one place we call `R CMD SHLIB`?